### PR TITLE
HUB-18986 Implement logs purging at regular interval in nginx

### PR DESCRIPTION
**Pull request by komalsodha**
Adding script to trigger logrotate once a day to purge nginx logs.  Rotation is based on size of 100 MB and max no of files kept is 2.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -34,11 +34,14 @@ RUN set -e \
...
     		tzdata \
-    && mkdir -p $WEBSERVER_HOME/bin $WEBSERVER_HOME/security \
+    		logrotate \
+    && mkdir -p $WEBSERVER_HOME/bin $WEBSERVER_HOME/security $WEBSERVER_HOME/logrotate \
 	&& rm "/etc/nginx/conf.d/default.conf" "/usr/bin/nc" "/var/log/nginx/error.log" "/var/log/nginx/access.log" \
     && chmod -R g+w $WEBSERVER_HOME \
     && chgrp -R 0 "/var/cache/nginx/" \
     && chmod -R 775 "/var/log/nginx" "/var/cache/nginx/" "/var/run" "/etc/nginx" \
+    && chown nginx:root $WEBSERVER_HOME/logrotate \
+    && chmod 777 $WEBSERVER_HOME/logrotate \
```
Comment (blackduck-cookt) [inline Dockerfile]: Does this really need to be world-writable?

File: logrotate.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+while true; do
+/usr/sbin/logrotate -s /opt/blackduck/hub/webserver/logrotate/logrotate.status -l /opt/blackduck/hub/webserver/logrotate/logrotate.log /etc/logrotate.d
```
Comment (blackduck-cookt) [inline logrotate.sh]: You might want to do something with stdout and stderr here.

File: logrotate.nginx.config [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+	create	
+	missingok
+	notifempty
+	rotate 2
+	sharedscripts
+	size 100m
+	postrotate
+    	[ ! -f /var/run/nginx.pid ] || kill -USR1 `cat /var/run/nginx.pid`
+  	endscript
+}
```
Comment (blackduck-cookt) [inline logrotate.nginx.config]: Does this mean nginx will be rotating logs itself?  If so, why do we need logrotate?

Comment (komalsodha) [reply logrotate.nginx.config]: Nope. Probably file name is misleading. This is logrotate's config which is run by background script.

**Review by blackduck-cookt — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (komalsodha) [reply Dockerfile]: Nope, I'll change it to 775.

**Comment by aclairekeum** [comment]
Was this targeted for the march release?

**Comment by komalsodha** [comment]
@aclairekeum  No.

File: logrotate.nginx.config (hunk 3 above)
Comment (aclairekeum) [reply logrotate.nginx.config]: then should we make it logrotate.config?

**Pull request merged**